pacman/utilities/json_utils.py

```diff
@@ -330,6 +331,10 @@ def graph_from_json(json_dict):
         vertex_add_contstraints_from_json(j_vertex, graph)
     for j_edge in json_dict["edges"]:
         edge = edge_from_json(j_edge, graph)
+        if not graph.outgoing_partition_exists(edge.pre_vertex, "JSON_MOCK"):
```

Comment:
WTF is test code doing in a non-test class?

Reply:
i have absolutely no idea.........